runtime/src/launch/slurm-gasnetrun_common/slurm-gasnetrun_common.h

```diff
@@ -282,6 +291,8 @@ static char* chpl_launch_create_command(int argc, char* argv[],
       len += sprintf(iCom+len, "--%s ", nodeAccessStr);
     if (walltime)
       len += sprintf(iCom+len, "--time=%s ", walltime);
+    if (nodelist)
+      len += sprintf(iCom+len, "--nodelist=%s ", nodelist);
```

Comment:
The command line we're building here ends up being executed by a shell, so we should quote the nodelist in case it contains characters that would be interpreted as shell syntax, such as '[]'.

Reply:
It's definitely an issue, and in the original code too. But admittedly it's extremely unlikely users will run into it.

```
[gbt@cflcs01:~] echo --nodelist=amd-000[1-7]
--nodelist=amd-000[1-7]
[gbt@cflcs01:~] touch -- --nodelist=amd-000{1,2,3,4,5,6,7}
[gbt@cflcs01:~] echo --nodelist=amd-000[1-7]
--nodelist=amd-0001 --nodelist=amd-0002 --nodelist=amd-0003 --nodelist=amd-0004 --nodelist=amd-0005 --nodelist=amd-0006 --nodelist=amd-0007
```

Reply:
The user may well have quoted it, but if they did the shell would have removed the quotes before we ever saw the arg. In any case, how does the user know we're going to pass the string they gave us to a shell (where shell meta-chars would be interpreted) rather than to an `exec*()` call where they would not?